lib/Indexer/Model/Record/HasFlagsTrait.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Phpactor\Indexer\Model\Record;
+
+trait HasFlagsTrait
+{
+    protected int $flags = 0;
+
+    public function setFlags(int $flags): self
+    {
+        $this->flags = $flags;
+        return $this;
+    }
+
+    public function addFlag(int $flag): self
+    {
+        $this->flags = $this->flags | $flag;
+        return $this;
+    }
+
+    public function flags(): int
+    {
+        return $this->flags;
+    }
+}
```

Comment:
as said, let's move this to class record for now. the valid flags are dependent on the tpye of record, and we may never add more.